### DISCUSSION TITLE
[A-L] (18 giugno 2024) Esercizio 4

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/progettazione-di-algoritmi#94] [Esami A-L]
Dare lo pseudocodice di un algoritmo $\Large O(n)$ che prende in input un array $\Large A[]$ di lunghezza $\Large n$ di numeri reali e trovi la differenza massima tra due elementi di $\Large A$.

Per esempio, se l'input è $\Large A = [1,\ 4,\ -5,\ 2,\ 3.5,\ 9]$, l'output dovrebbe essere $\Large 14=9-(-5)$

**Comment by Fastonez** [comment]
Ho provato questo codice con python e rispetto all'esempio funziona. Fatemi sapere se potrebbero esserci casi particolari per cui potrebbe non funzionare. 
Il costo è in O(n), infatti esegue un solo ciclo for pari alla lunghezza dell'array A.

def es3(P):
    minimo = 0;
    massimo = 0;
    diff_max = 0;

    for item in range(0, len(P)):
        if P[item] > massimo:
            massimo = P[item]
        elif P[item] < minimo:
            minimo = P[item]
        diff_max = abs(massimo - minimo)
   
    return diff_max

**Comment by IonutCicio** [comment]
```rust
fn max_diff(array: &[isize]) -> isize {
    array.iter().max().unwrap() - array.iter().min().unwrap()
}
```

**Comment by daromacarlo** [comment]
![immagine](https://github.com/user-attachments/assets/84d3743c-3e76-47b0-9f5e-b5f4eae78fc5)

**Comment by Near047** [comment]
`
fun(A):

 min = +inf
 max = -inf 
 for v in A:
  if v >= max:
   max = v
  if v <= min:
   min = v
 end_for
 if min < 0 And max >= 0:
  return max -(min)
 else:
  return abs(max) - abs(min)
`
